### DISCUSSION TITLE
Using a dict as a reactive

### DISCUSSION THREAD

**Discussion by mattloose** [Textualize/textual#2538] [Q&A] [answered]
I'm sorry if this is answered or explained elsewhere.

I'd like to use a dict as a reactive in a widget. If I pass in a dict with keys already existing, the reactivity works fine. However, if I add a new key to the dict it does not update as I might expect.

Any pointers on what one might do?

Thanks for a great tool!

**Comment by davep** [comment, marked as answer]
You may find https://github.com/Textualize/textual/discussions/2193 helpful here.

A fairly useless but hopefully illustrative example could be:

```python
from textual.app      import App, ComposeResult
from textual.reactive import reactive
from textual.widgets  import Header, Footer, Label

class DictReactiveApp( App[ None ] ):

    my_dict = reactive(dict(), always_update=True, init=False)

    BINDINGS = [
        ( "space", "change", "" ),
    ]

    def compose( self ) -> ComposeResult:
        yield Header()
        yield Label(str(self.my_dict))
        yield Footer()

    def watch_my_dict(self) -> None:
        self.query_one(Label).update(str(self.my_dict))

    def action_change( self ) -> None:
        self.my_dict[len(self.my_dict)] = len(self.my_dict)
        self.my_dict = self.my_dict

if __name__ == "__main__":
    DictReactiveApp().run()
```

Note that making the `reactive` `always_update=True` and then assigning it to itself when you want to signal it's changed is the key here.

**Comment by mattloose** [comment]
Thank you!

I'll try this out.
